### PR TITLE
Add font command

### PR DESCRIPTION
**Pull request by bcr**
I think I have all the party supplies I need in the `freetype-py` package. This should run on Linux if you want to give it a whirl @FoamyGuy

### CODE REVIEW

**Comment by FoamyGuy** [comment]
image
It does make the 'e' like you mentioned

**Comment by bcr** [comment]
Good deal, I think that's the only technical hurdle I had. I should be able to output glyphs to a BDF no problem then.

File: commands/font.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    file.write("CHARS\n")
+
+def output_footer(file):
+    file.write("ENDFONT\n")
+
+def do_font(args):
+    logging.debug("Making a font from %s %d" % (args.fontpath, args.fontsize))
+    font = fontutil.Font(args.fontpath, args.fontsize)
+    filename = make_suggested_filename(font, args.fontsize)
+    chars_to_output = "Hello, world!"
```
Comment (bcr) [inline commands/font.py:56]: I need to come up with some way to specify the characters. Not sure exactly what to do. All of ASCII?

**Comment by bcr** [comment]
So this is working pretty good. There is one slight letter spacing problem that I need to figure out -- if I output `jumps` with the font I got from `python3 .\blinka.py font --fontpath C:\Windows\Fonts\ariali.ttf --fontsize 23`, the 'j' and the 'u' are a little crowded.

**Comment by FoamyGuy** [comment]
Is there a way to specify which characters get included in the bdf file? 

I tested this out and the resulting font does work and look correct, but it only included a few characters. Any others I attempt to show get ignored.

**Review by FoamyGuy — APPROVED** [review]
Ah, I see now you mentioned already about a way to specify the characters whoops, missed that before.

I tried this out on a few different fonts and sizes and everything was working well.

**Comment by bcr** [comment]
Good deal. Yeah, so there's a little bit of a letter spacing issue that can get fixed later. And then yeah, what should we do about the charset (what glyphs to include). Part of the current problem is that we're generating a whole bunch of characters no one is using, and even if CP is selectively loading those, the BDF still has them and they're taking up space.

**Comment by FoamyGuy** [comment]
I think it'd be great to have a few options: 
- all of ascii
- user selects characters with a quoted parameter on the command line. (most useful for very small sub sets like using "cm" when the font will only be used for a units label.
- user selects characters by specifying another file as an input parameter. That file can contain plain text of whatever characters are wanted. That is helpful when they want a large but not complete subset.

**Comment by bcr** [comment]
For a first cut I think I'm going with:

* `string.printable`
* Specify characters via `--chars`
* Specify "all characters in this UTF-8 encoded file" via `--chars-file`

**Pull request merged**